src/dbally/utils/gradio_adapter.py

```diff
@@ -0,0 +1,173 @@
+import sys
+from typing import Dict, List, Optional, Tuple
+
+import gradio
+import pandas as pd
+
+from dbally import BaseStructuredView
+from dbally.collection import Collection
+from dbally.prompts import PromptTemplateError
+from dbally.similarity import SimilarityIndex
+from dbally.utils.errors import NoViewFoundError, UnsupportedQueryError
+from dbally.utils.log_to_file import FileLogger
+
+CONSOLE_FILE_NAME = "console.log"
+sys.stdout = FileLogger(CONSOLE_FILE_NAME)
+
+
+class GradioAdapter:
+    """
+    A class to adapt and integrate data collection and query execution with Gradio interface components.
+    """
+
+    def __init__(self, preview_limit: int = 20):
+        """
+        Initializes the GradioAdapter with a preview limit.
+
+        Args:
+            preview_limit: The maximum number of preview data records to display. Default is 20.
+        """
+        self.preview_limit = preview_limit
+        self.similarity_store_list = []
```

Comment:
we don't need to store any information about similarity stores. That is already handled by Collection. Also updating similarity stores should not happen inside gradio adapter.